### PR TITLE
Add FMIActuatorPluginKnownInputTest

### PR DESCRIPTION
**Pull request by traversaro**
For now, this test just simulates the pendulum + FMU transmission
with a constant input, and saves the result in .mat file to simplify the inspection. 

It is already useful on its own for developers, but the idea
is that it can be extended to more inputs (such as as sine) and to
check the behavior of simple models + transmission against their nominal behavior.

### CODE REVIEW

**Comment by traversaro** [comment]
For example, this is how I verified that the acceleration feedback for https://github.com/robotology/gazebo-fmi/issues/17 had the correct sign: 
ode

Note that there some strange spikes in acceleration that I guess are some kind of numerical integration artifacts due to ODE (probably it is possible to get rid of them by tuning the physics parameters as described in http://gazebosim.org/tutorials?tut=physics_params&branch=physics_params . It would be interested anyhow to investigate them, as the system is extremely simple and such spike are quite surprising/scary, but this is out of the scope of this PR.

File: plugins/actuator/GazeboFMIUtils.hh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{
+    if (jointPtr->GetType() & gazebo::physics::Base::HINGE_JOINT)
+    {
+        // Compute joint acceleration
+        // For a link `L`, the method ignition::math::Vector3d WorldAngularAccel () const
+        // returns the angular acceleration \$ {}^A \omega_{A,L} \$.
+        // Given a hinge/revolute joint ${P, C}$ connecting the links $P$ (parent) and $C$ (child), the method
+        // virtual ignition::math::Vector3d GlobalAxis( unsigned int  _index) const
+        // returns the angular part of the joint motion subspace \$  ^A s_{P,C} \in \mathbb{R}^6  \$
+        // that we indicate with \$ {}^A a_{P, C} \in \mathbb{R}^3 \$.
```
Comment (prashanthr05) [inline plugins/actuator/GazeboFMIUtils.hh]: we indicate with  $ {}^A axis_{P, C} \in \mathbb{R}^3 $ according to the code.

File: plugins/actuator/GazeboFMIUtils.hh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        // \ddot{\theta} =  \left( ^C s_{P,C} \right)^T  {}^C \dot{\omega}\_{P,C} =  \left( ^C s_{P,C} \right)^T  ( {}^C \dot{\omega}\_{A,C} - {}^C \dot{\omega}\_{A,P} )  =
+        // \left( ^A s_{P,C} \right)^T ( {}^A \dot{\omega}\_{A,C} - {}^A \dot{\omega}\_{A,P} )
+        // \$
+
+        ignition::math::Vector3d A_axis_P_C = jointPtr->GlobalAxis(0u);
+        gazebo::physics::LinkPtr parent = jointPtr->GetParent();
+        gazebo::physics::LinkPtr child = jointPtr->GetChild();
+        ignition::math::Vector3d A_domega_A_P = parent->WorldAngularAccel();
+        ignition::math::Vector3d A_domega_A_C = child->WorldAngularAccel();
+        return A_axis_P_C.Dot(A_domega_A_C-A_domega_A_P);
```
Comment (prashanthr05) [inline plugins/actuator/GazeboFMIUtils.hh]: can we do a little comparison here of this quantity with Gazebo's ground truth of the joint acceleration ? is it possible ?

Comment (traversaro) [reply plugins/actuator/GazeboFMIUtils.hh]: There is no ground truth in Gazebo. In theory it is possible to do this comparison by post processing the data saved in the `.mat` files.

**Comment by traversaro** [comment]
I added a test using a [`Modelica.Blocks.Nonlinear.FixedDelay`](https://doc.modelica.org/om/Modelica.Blocks.Nonlinear.FixedDelay.html) block, and the tests are now faiing because the resulting FMU exported from OpenModelica is actually generating a [`NaN`](https://en.wikipedia.org/wiki/NaN), which I guess it is an OpenModelica problem.

**Comment by traversaro** [comment]
I tried to inspect the `DelayTransmission.fmu` (generated in `build/plugins/actuator/test`) with the [FMPy gui](https://github.com/CATIA-Systems/FMPy) (`python3 -m fmpy.gui`) and I obtained a strange error:
~~~
straversaro@iiticublap103:~/src/gazebo-fmi/build/plugins/actuator/test$ python3 -m fmpy.gui
JSON object expected ',' or '}', got: "lineEnd":3,"colStar
Aborted (core dumped)
~~~ 
I discovered that FMUs generated by OpenModelica have a bug (https://trac.openmodelica.org/OpenModelica/ticket/4020) and are sensitive to the system's locale, a problem similar to https://github.com/robotology/idyntree/issues/288 .
Indeed launching `LANG=en_US.utf8 LC_ALL=C.UTF-8 python3 -m fmpy.gui` everything works fine. Interesting, I see no `NaN` in the output

I tried to launch the tests in this PR as: 
`LANG=en_US.utf8 LC_ALL=C.UTF-8 ctest -VV -R Known`, but the problem is still there, so this is probably not the problem. I am starting to wonder if the problem is actually in the FMILibrary or in how  we use it. 
If someone has a Simulink license, it would be interesting to try to load the `DelayTransmission.fmu` model (see https://it.mathworks.com/help/simulink/in-product-solutions.html ) in Simulink and try a couple simulations.

**Comment by traversaro** [comment]
@prashanthr05 for some reason I am unable to reply to your comment https://github.com/robotology/gazebo-fmi/pull/29#discussion_r218817254 . I addressed it in https://github.com/robotology/gazebo-fmi/pull/29/commits/5fd1da17185c362a667d9cbacde5e755cf5c074f .

**Comment by traversaro** [comment]
The tests with the DelayTransmission are working as intended in OpenModelica 1.14.0~dev-375-ge9bb8d6, so the problem with DelayTransmission was probably a OpenModelica specific problem.

**Comment by traversaro** [comment]
For some reason in Travis there is still the problem of the Delay test, I have disabled it in https://github.com/robotology/gazebo-fmi/pull/29/commits/abe72f31b07628258c05b391d412dca66a9cbdd8 and I opened a new issue for it in https://github.com/robotology/gazebo-fmi/issues/58 .

**Comment by traversaro** [comment]
cc  @prashanthr05

**Comment by traversaro** [comment]
@prashanthr05 merging as this are just tests, feel free to comment later.

**Comment by prashanthr05** [comment]
> For some reason in Travis there is still the problem of the Delay test, I have disabled it in [abe72f3](https://github.com/robotology/gazebo-fmi/commit/abe72f31b07628258c05b391d412dca66a9cbdd8) and I opened a new issue for it in #58 .

This seems like a strange issue given the fact that the test passes locally.

**Comment by traversaro** [comment]
> This seems like a strange issue given the fact that the test passes locally. 

We unfortunately still use the Nightly build of OpenModelica, so it is possible that the version I have on my laptop is different from the one in Travis. I think we can wait for OpenModelica 1.14  stable release and check if we can switch to that.

**Pull request merged**